File: LAB2/Task2.m
% KUMAIL RAZA
% BESE 5
% DIP LAB 2
% TASK 3

A = imread('lab2-task3-image.jpg');
% RED TO YELLOW
red = A(:,:,1); green = A(:,:,2); blue = A(:,:,3);

%// Change here
mred = red >= 150 & red <= 255; 
mgreen = green >= 0 & green <= 100; 
mblue = blue >= 0 & blue <= 100;

%// Back to before
final_mask = mred & mgreen & mblue;
red(final_mask) = 255; green(final_mask) = 255; blue(final_mask) = 0;
A = cat(3, red, green, blue);

% BLUE TO MAGENTA
red = A(:,:,1); green = A(:,:,2); blue = A(:,:,3);
mred = red >= 0 & red <= 100; 
mgreen = green >= 0 & green <= 100; 
mblue = blue >= 150 & blue <= 255;

%// Back to before
final_mask = mred & mgreen & mblue;
red(final_mask) = 255; green(final_mask) = 0; blue(final_mask) = 255;
A = cat(3, red, green, blue);

% GREEN TO CYAN
red = A(:,:,1); green = A(:,:,2); blue = A(:,:,3);
mred = red >= 0 & red <= 50; 
mgreen = green >= 150 & green <= 255; 
mblue = blue >= 0 & blue <= 100;

%// Back to before
final_mask = mred & mgreen & mblue;
red(final_mask) = 0; green(final_mask) = 255; blue(final_mask) = 255;
A = cat(3, red, green, blue);

imshow(A);

File: LAB2/Task2.py
# Kumail Raza
# BESE 5A
# DIP Lab 2 
# Task 2 Python

from PIL import Image
img = Image.open('lab2-task3-image.jpg')
pixels = img.load() # create the pixel map

for i in range(img.size[0]): # for every pixel:
    for j in range(img.size[1]):
    	red = pixels[i,j][0]
    	green = pixels[i,j][1]
    	blue = pixels[i,j][2]

    	if red in range(100,255) and green in range(0,100) and blue in range(0,100):
        # if pixels[i,j] == (range(150,255), range(0,100), range(0,100)):
        	pixels[i,j] = (255, 255 ,0)

        if red in range(0,100) and green in range(0,100) and blue in range(150,255):
        # if pixels[i,j] == (range(150,255), range(0,100), range(0,100)):
        	pixels[i,j] = (255, 0 ,255)
        if red in range(0,50) and green in range(150,255) and blue in range(0,100):
        # if pixels[i,j] == (range(150,255), range(0,100), range(0,100)):
        	pixels[i,j] = (0, 255 ,255)
          
        

img.show()
        